iOS Client/Reducers/MBAppReducer.swift

```diff
@@ -59,8 +65,27 @@ func devotionReducer(action: Action, state: DevotionState?) -> DevotionState {
     switch action {
     case let action as LoadedDevotions:
         nextState.devotions = action.devotions
+    case let action as SelectedDevotion:
+        nextState.selectedDevotion = action.devotion
+        if case Loaded.loaded(data: let devotions) = nextState.devotions {
+            let newDevotions: [LoadedDevotion] = devotions.map { devotion in
+                if devotion.date == nextState.selectedDevotion?.date {
+                    return LoadedDevotion(devotion: devotion.devotion, read: true)
+                } else {
+                    return devotion
+                }
+            }
+            nextState.devotions = Loaded.loaded(data: newDevotions)
+        }
```

Comment:
since we are dealing with classes (ref types), I think it would work to delete lines 70-79 and just write `nextState.selectedDevotion.read = true`

Reply:
pretty sure these are all structs though. DevotionState is a struct and devotions are structs

Reply:
Made this more succinct in the latest commits